Add planned enhancement for weight and more specific error messages for optional fields (#307)
AY2324S2-CS2103T-T17-3/tp: docs/DeveloperGuide.md

    1. _{explain how to simulate a missing/corrupted file, and the expected behavior}_

1. _{ more test cases …​ }_

## Appendix: Planned Enhancements
FitBook's team size is 5.

1. **Allow for more flexible weight management:** The current implementation of tracking clients' weights only allows users to add, modify or delete the latest weight value at the current date and time. We plan to make this feature more flexible by allowing users to add/modify a client's weight at a specified date and time of their choice.

> 1. **Provide more specific error messages:** When removing optional fields of a client, we should provide more detailed error messages. This is currently only available for the `weight` field, where removing a weight value from a client that has no weight value associated with them prompts the error message, `There are no more weight values to be removed. This client has no more weight values associated with them.`. We plan to provide more specific error messages when a client edits a field that has not changed / has nothing to remove, where the error message will be similar to the one seen in the `weight` field.

Author: ```suggestion
1. **Provide more specific error messages:** When removing optional fields of a client, we should provide more detailed error messages. This is currently only available for the `weight` field, where removing a weight value from a client that has no weight value associated with them prompts the error message, `There are no more weight values to be removed. This client has no more weight values associated with them.`. We plan to provide more specific error messages when a client edits a field that has not changed / has nothing to remove, where the error message will be similar to the one seen in the `weight` field.

```

> 1. **Provide more specific error messages:** When removing non-optional fields of a client, we should provide more detailed error messages. This is currently only available for the `weight` field, where removing a weight value from a client that has no weight value associated with them prompts the error message, `There are no more weight values to be removed. This client has no more weight values associated with them.`. We plan to provide more specific error messages to the other optional fields of a client , where the error message will be similar to the one seen in the `weight` field.

Author: ```suggestion
1. **Provide more specific error messages:** When removing optional fields of a client, we should provide more detailed error messages. This is currently only available for the `weight` field, where removing a weight value from a client that has no weight value associated with them prompts the error message, `There are no more weight values to be removed. This client has no more weight values associated with them.`. We plan to provide more specific error messages when a client edits a field that has not changed / has nothing to remove, where the error message will be similar to the one seen in the `weight` field.
```

> 1. **Provide more specific error messages:** When removing non-optional fields of a client, we should provide more detailed error messages. This is currently only available for the `weight` field, where removing a weight value from a client that has no weight value associated with them prompts the error message, `There are no more weight values to be removed. This client has no more weight values associated with them.`. We plan to provide more specific error messages to the other non-optional fields of a client , where the error message will be similar to the one seen in the `weight` field.

Author: ```suggestion
1. **Provide more specific error messages:** When removing non-optional fields of a client, we should provide more detailed error messages. This is currently only available for the `weight` field, where removing a weight value from a client that has no weight value associated with them prompts the error message, `There are no more weight values to be removed. This client has no more weight values associated with them.`. We plan to provide more specific error messages to the other optional fields of a client , where the error message will be similar to the one seen in the `weight` field.
```
This is a typo!
